internal/internal_workflow.go

```diff
@@ -1182,16 +1194,14 @@ func (d *dispatcherImpl) Close() {
 	}
 	d.closed = true
 	d.mutex.Unlock()
-	// This loop breaks our expectation that only one workflow
-	// coroutine is running at any time because it triggers all workflow goroutines
-	// to call their defers at once. Adding synchronization seemed more problematic because
-	// it could block eviction if there was a deadlock.
-	for i := 0; i < len(d.coroutines); i++ {
-		c := d.coroutines[i]
-		if !c.closed.Load() {
-			c.exit()
+	// We need to exit the coroutines in a separate goroutine because:
```

Comment:
So to confirm, `exit` has always been non-blocking? Or was the send `c.unblock` blocking (even though it obviously didn't wait for code to complete)? Can you confirm that returning from dispatcher `Close()` before the `exit`/`unblock`s have even been sent is safe?

Reply:
>So to confirm, exit has always been non-blocking

Yes

> Can you confirm that returning from dispatcher Close() before the exit/unblocks have even been sent is safe?

It should be as safe as returning after they are sent, but not processed (the old behaviour)